server/app/models/stack.rb

```diff
@@ -11,6 +11,8 @@ class Stack
   has_many :stack_revisions, dependent: :destroy
   has_many :stack_deploys, dependent: :destroy
   has_many :grid_services
+  has_many :volumes
+  embeds_many :external_volumes
```

Comment:
@SpComb I was actually going to propose similar solution.. I think this is the simplest way to go?